telegram/src/main/kotlin/com/github/kotlintelegrambot/entities/inputmedia/InputMedia.kt

```diff
@@ -7,28 +7,55 @@ import com.google.gson.annotations.SerializedName
  * Represents the content of a media message to be sent.
  * https://core.telegram.org/bots/api#inputmedia
  */
-sealed class InputMedia {
-    abstract val type: String
-    abstract val media: TelegramFile
-    abstract val caption: String?
-    abstract val parseMode: String?
+sealed interface InputMedia {
+    val type: String
+    val media: TelegramFile
+    val caption: String?
+    val parseMode: String?
 }
 
 /**
  * Interface to mark all the media types that can be sent within a group of media for
  * operations like `sendMediaGroup`.
  */
-interface GroupableMedia
+sealed interface GroupableMedia : InputMedia
 
 class MediaGroup private constructor(val medias: Array<out GroupableMedia>) {
     init {
-        if (!(2..10).contains(medias.size)) {
-            throw IllegalArgumentException("media groups must include 2-10 items")
-        }
+        require(medias.size in 2..10) { "media groups must include 2-10 items" }
     }
 
     companion object {
-        fun from(vararg media: GroupableMedia): MediaGroup = MediaGroup(media)
+        fun from(vararg media: GroupableMedia): MediaGroup = MediaGroup(generateUniqueNamesForNamelessByteArrays(media))
+
+        private fun generateUniqueNamesForNamelessByteArrays(media: Array<out GroupableMedia>): Array<GroupableMedia> {
```

Comment:
I'm wondering if we should push the decision about the name for the `ByteArray` files to consumers of the lib. instead (i.e. making `filename` not optional in `TelegramFile.ByByteArray`). It feels like they are in a better position to make the right call here, and not forcing it in the API may lead to unexpected results (e.g. they may not know that if they don't put a name in their `ByByteArray` objects, a random `nonamex` or whatever will be used instead). And if they don't care about the name, they can use the default of of their choice (or we could even provide a default if we think that will be a common use case). What do you think?

Reply:
Making `filename` not optional will be a breaking change. So, if we want to make it non-nullable, we still need to provide some default (but now non-null) value (strictly speaking, it will be breaking change anyway, but only for someone who explicitly passed `null` for this parameter, repeating current default).
Not sure why I thought that Telegram API requires filenames uniqueness across `MediaGroup`. Revised their docs, but haven't found it. So, yes, we could provide some non-null default in TelegramFile.ByByteArray constructor (and not worry about filename collision if there will be two `TelegramFile`s with the same default value in `MediaGroup`) - it will greatly simlify the code and indeed will make interaction with API more straightforward.

While revising docs for [sendMediaGroup](https://core.telegram.org/bots/api#sendmediagroup) found another restriction, we're missing check for: 

>Documents and audio files can be only grouped in an album with messages of the same type.

Will implement it the scope of this PR, because this very thing happens in the tests I've added. 🙈🙈🙈